### PR TITLE
Accept args and no-cache for run

### PR DESCRIPTION
**Pull request by DuBistKomisch**
Fixes https://github.com/buildkite-plugins/docker-compose-buildkite-plugin/issues/197, and threw in `no-cache` while I was at it since it's theoretically useful, but left out `build-parallel` since `run` only accepts one service anyway.

Wrote some tests here but also tested with our project and it's working.

Let me know if I've missed anything I need to change, I just grepped for the configuration property names.

`build_params` construction could probably be a shared method but oh well.

### CODE REVIEW

File: plugin.yml [hunk 1]
```diff
@@ -89,10 +89,11 @@ configuration:
...
     leave-volumes: [ run ]
-    no-cache: [ build ]
+    no-cache: [ build, run ]
     use-aliases: [ run ]
     dependencies: [ run ]
     ansi: [ run ]
     tty: [ run ]
     workdir: [ run ]
     user: [ run ]
+    args: [ build, run ]
```
Comment (DuBistKomisch) [inline plugin.yml]: this didn't exist and these don't seem to be in any particular order, so I just chucked it on the end here

**Comment by jayco** [comment]
Hey, @DuBistKomisch thanks for sending this through, sorry it has taken a while to take a look at this.

As per https://github.com/buildkite/agent/pull/1161 the current agent version won't run multiple docker-compose plugin invocations in a single step. We're investigating that at the moment and once that's resolved we will come back to get this. Thanks for your patience!

**Comment by DuBistKomisch** [comment]
@jayco I'm sorry, but I don't see how that relates to this PR?

**Comment by jayco** [comment]
@DuBistKomisch Oh my mistake, I was looking at the example in the original issue and thought this might run into issues with multiple docker-compose invocations. This looks good to merge!

**Comment by jayco** [comment]
@DuBistKomisch sorry we had to revert this we will get this fixed and merged.

**Comment by DuBistKomisch** [comment]
no worries :)

**Pull request merged**